src/main/java/net/masterthought/cucumber/json/Match.java

```diff
@@ -1,13 +1,55 @@
 package net.masterthought.cucumber.json;
 
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.LinkedList;
+import java.util.List;
+
 public class Match {
 
     // Start: attributes from JSON file report
-    private final String location = null;
+    private final JsonElement location = null;
     // End: attributes from JSON file report
 
     public String getLocation() {
-        return location;
+        if (location==null){
+            return "";
+        }
+        if (location.isJsonPrimitive() && location.getAsJsonPrimitive().isString()) {
+            return location.getAsString();
+        } else {
+
+            String filename = location.getAsJsonObject().get("filepath").getAsJsonObject().get("filename").getAsString();
+
+            LinkedList<Integer> lineList = new LinkedList<>();
+            JsonArray lineArray = location.getAsJsonObject().get("lines").getAsJsonObject().get("data").getAsJsonArray();
+            for (JsonElement lineElement : lineArray){
+                lineList.add(lineElement.getAsInt());
+            }
+
+            StringBuilder sb = new StringBuilder();
+            sb.append(filename);
+            sb.append(":");
+            if (lineList.size()>1 && areConsecutive(lineList)){
+                sb.append(lineList.getFirst());
+                sb.append("..");
+                sb.append(lineList.getLast());
+            } else {
+              sb.append(StringUtils.join(lineList, ','));
+            }
+            return sb.toString();
+        }
     }
 
+    private boolean areConsecutive(List<Integer> lines){
+        int firstLine = lines.get(0);
+        for (int i=1; i<lines.size(); i++){
+            if (lines.get(i)!=firstLine+i){
```

Comment:
the intention is to check if we can replace lines with a range (f.e 1..10 instead of 1,2,3,4,5,6,7,8,9,10)